### PR TITLE
Update instructions to point to pip install

### PR DESCRIPTION
**Pull request by hawkowl**

### CODE REVIEW

**Comment by hawkowl** [comment]
Fixes https://github.com/matrix-org/synapse/issues/1224

File: .circleci/merge_base_branch.sh [hunk 1]
```diff
@@ -9,13 +9,16 @@ source $BASH_ENV
 
 if [[ -z "${CIRCLE_PR_NUMBER}" ]]
 then
-    echo "Can't figure out what the PR number is!"
-    exit 1
+    echo "Can't figure out what the PR number is! Assuming develop."
```
Comment (richvdh) [inline .circleci/merge_base_branch.sh]: this looks unrelated, though useful.

I would s/Assuming develop/Assuming it will be based on develop/ though.

File: README.rst [hunk 2]
```diff
@@ -533,9 +512,7 @@ fix try re-installing from PyPI or directly from
 
     # Install from PyPI
     pip install --user --upgrade --force pynacl
-
-    # Install from github
-    pip install --user https://github.com/pyca/pynacl/tarball/master
```
Comment (richvdh) [inline README.rst]: synapse != pynacl

Comment (hawkowl) [reply README.rst]: oh

File: README.rst [hunk 3]
```diff
@@ -459,7 +459,7 @@ Troubleshooting
 Troubleshooting Installation
 ----------------------------
 
-Synapse requires pip 1.7 or later, so if your OS provides too old a version you
+Synapse requires pip 8 or later, so if your OS provides too old a version you
```
Comment (richvdh) [inline README.rst]: 1.7lol

Comment (hawkowl) [reply README.rst]: 1.7, ahh, such a classic pip release (which literally won't install packages now)
